test-data/unit/check-attr.test

```diff
@@ -74,8 +74,8 @@ A(1, [2], '3', 4, 5)  # E: Too many arguments for "A"
 [case testAttrsUntypedNoUntypedDefs]
 # flags: --disallow-untyped-defs
 import attr
-@attr.s    # E: Function is missing a type annotation for one or more arguments
-class A:
+@attr.s
+class A:  # E: Function is missing a type annotation for one or more arguments
```

Comment:
Hm... Do I understand correctly that this is a backwards incompatibility? If the user had a `# type: ignore` on the decorator it would no longer work, right? I can't even figure out which part of this PR moved the line where the error occurs. :-(

(FWIW I agree that this is a better place for the error to occur. But I am concerned about compatibility.)

Reply:
It shouldn't be a backwards incompatibility. Even though the `lineno` of the error is different (648 and 651 of `fastparse.py`, since you were curious), we use the same `end_lineno` trick that's used for functions to respect `# type: ignore` comments occurring at the old location, _even for versions prior to 3.8_. This is done in `fastparse.py`, line 649 (for < 3.8) and line 652 (for >= 3.8).